base/logging.jl

```diff
@@ -169,9 +169,6 @@ overridden:
 
 There's also some key value pairs which have conventional meaning:
 
-  * `progress=fraction` should be used to indicate progress through an
```

Comment:
why take this out?

Reply:
Yes, I loved this feature and removing it makes me sad.  But I think it's too unreliable as a user facing interface for progress reporting.  Consider

```julia
for i=1:N
    foo()
    @info "blah" progress=i/N
end
```

This looks super simple and nice, but what if `foo()` throws?  Then you're left with an incomplete progress bar which needs to be cleaned up with a heuristic of some kind.

We might "fix" this by limiting things to one progress bar at a time. But this doesn't make sense for nested progress bars and more general concurrent situations where the loop above may be executing from more than one task at once.  In these cases you really want a reliable way to detect when the progress step is complete.

So I think it's better to remove this for now and reintroduce a more reliable progress reporting interface later.

Reply:
I think that's a sane approach. Better to be a little cautious now and give the good stuff later.